assets/js/plugin-check-admin.js

```diff
@@ -0,0 +1,65 @@
+( function ( data ) {
+	const checkItButton = document.getElementById( 'plugin-check__submit' );
+	const pluginsList   = document.getElementById( 'plugin-check__plugins-dropdown' );
+
+	// Return early if the elements cannot be found on the page.
+	if ( ! checkItButton || ! pluginsList ) {
+		console.error( 'Missing form elements on page' );
+		return;
+	}
+
+	checkItButton.addEventListener( 'click', (e) => {
+		e.preventDefault();
+
+		const pluginCheckData = new FormData();
+
+		// Collect the data to pass along for generating a check results.
+		pluginCheckData.append( 'action', 'plugin_check_run_checks' );
+		pluginCheckData.append( 'nonce', data.nonce );
+		pluginCheckData.append( 'plugin', pluginsList.value );
+
+		fetch(
+			data.ajaxUrl,
+			{
+				method: 'POST',
+				credentials: 'same-origin',
+				body: pluginCheckData
+			}
+		)
+		.then(
+			( response ) => {
+				if ( ! response.ok ) {
+					throw new Error( `${response.status}: ${response.statusText}` );
+				}
```

Comment:
This here is what throws a wrench in things. You already error out due to the bad response code, which makes it impossible to even reach the code below where we expose the more specific JSON error that is present.

I think by removing this our error logging works correctly. Requests that come back with a bad response are automatically logged in the console anyway, so no need to separately log the status code in our logic.